test/unit/storage_subformat_vagrant_base_test.py

```diff
@@ -28,6 +29,29 @@ def setup(self):
         self.vagrantconfig.get_virtualsize = Mock(
             return_value=42
         )
+
+    def config_create_image_format_mocks(
+            self, mock_rand, mock_mkdtemp, mock_open
+    ):
+        """
+        Creates mocks to test the function ``create_image_format`` in child
+        classes of DiskFormatVagrantBase.
+
+        :return: mock return_value of `open()`, this can be used to retrieve
+            values that were written to ``open``'d files
+        """
+        mock_rand.return_value = 0xa
+        mock_mkdtemp.return_value = 'tmpdir'
+
+        mock_open.return_value = MagicMock(spec=io.IOBase)
+        file_handle = mock_open.return_value.__enter__.return_value
+
+        return file_handle
+
```

Comment:
So sorry but I don't like this concept. I think it does not belong here. What you are trying to do is to reduce code duplication which is great but the code here is out of scope and has no relation to the vagrant base class testing. So I see two things:

1. you want to define init values for the rand and mkdtemp mocks. This is nice an generically useful. So there is ```test/unit/test_helper.py``` which should be the location for such helpers. However I would see that one in the scope of consistency and not so much helping to reduce code duplication

2. You want to create the file_handle mock as return to the open mock. I like this as it is used very often. You could consider another method to test_helper to serve you better here

config_create_image_format_mocks that reads like "do something" and what something is can only be understood by reading the code. And why this code exists can only be understood by reading and understanding the test. imho we shouldn't do that